basket/main.py

```diff
@@ -165,7 +204,11 @@ def _has_package(self, package, version):
     def _download(self, package, version, url):
         self.downloaded_packages.append({'name': package, 'version': version})
         path = self.os.path.join(self.root, url[url.rfind('/') + 1:])
-        self.urllib.urlretrieve(url, path)
+        #self.urllib.urlretrieve(url, path)
+        file = self.urllib.urlopen(url)
+        out_save = open(path, 'wb')
+        out_save.write(file.read())
+        out_save.close()
```

Comment:
It looks like a re-implementation of `urlretrieve()`, which still exists in Python 3 (`urllib.request.urlretrieve`). I would say that we could keep using the default Python function?